src/actionCreators/resources.js

```diff
@@ -47,6 +48,24 @@ export const retrieveResource = (currentUser, uri) => (dispatch) => {
     })
 }
 
+// A thunk that publishes (saves) a new resource in Trellis
+export const publishResource = (currentUser, group) => (dispatch, getState) => {
+  dispatch(publishStarted()) // clears possible residual server error
+
+  const rdf = new GraphBuilder(getState().selectorReducer).graph.toCanonical()
+
+  return publishRDFResource(currentUser, rdf, group).then((result) => {
+    const resourceUrl = result.response.headers.location
+    dispatch(assignBaseURL(resourceUrl))
+    dispatch(showResourceURIMessage(resourceUrl))
+    // Need to regenerate RDF now that we have baseURL
+    const updatedRdf = new GraphBuilder(getState().selectorReducer).graph.toCanonical()
```

Comment:
There is going to be a race condition here between line 59 and line 62. If assignBaseURL hasn't completed in time, it will produce an incorrect MD5. I'd recommend putting the base URL in a local copy of state and pass that to the GraphBuilder.

Reply:
@justinlittman   I hear you ... but this is straight copied from what was in code -- this is a refactor PR.  Can we do the fix separately, or maybe merge this and put fix in your PR #1111?  🤞

Reply:
or maybe you can push a commit up to this branch if it's super easy for you?